pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -316,8 +324,10 @@ func (gc *GarbageCollector) processAttemptToDeleteWorker(ctx context.Context) bo
 	switch action {
 	case forgetItem:
 		gc.attemptToDelete.Forget(item)
+		gc.attemptToDeleteMetrics.Forget(item)
```

Comment:
doesn't the queue already have metrics?

Reply:
@deads2k  yes we do, but they are not usable due to the nature of the errors (https://github.com/kubernetes/kubernetes/pull/110329#issuecomment-1151561954): 

the exisitng metrics have following problems:

`workqueue_retries_total`:  it is not possible to distinguish between transient and persistent error in our scenario. No way to get number of retries (or time) per item
`workqueue_depth` and `workqueue_queue_duration_seconds`: does not work - since retried items are not really in the queue and thus are not registred via the metric and are just waiting to be added to the queue after a backoff
`workqueue_work_duration_seconds`: not usable - tracks only work duration
`workqueue_adds_total`: not usable - tracks only adds
`workqueue_unfinished_work_seconds`: not usable (usually 0)
`workqueue_longest_running_processor_seconds`: not usable (usually 0)